Create element.py

valeriereyes5 opened the pull request:
Created Element Class, Implemented constructor and toString

valeriereyes5 committed:
Create element.py

Created Element Class, Implemented constructor and toString

isaacrivriv commented on src/element.py:
```diff
@@ -0,0 +1,49 @@
+            concat = concat + key+ "=" + str(value)+ ","
+        print(concat)
+
+El = Element('10')
```
If it works ok we just need to remove these little tests you have here

isaacrivriv commented on src/element.py:
```diff
@@ -0,0 +1,49 @@
+        self.precise = self.element_dic['precise']
+        self.type = self.element_dic['type']
+
+    def toString(self):
```
python has a built in to string method you just need to overload it. I believe its the __str__ function, We need to change the  name here

isaacrivriv commented on src/element.py:
```diff
@@ -0,0 +1,49 @@
+        concat = ''
+        for key,value in self.element_dic.iteritems():
+            concat = concat + key+ "=" + str(value)+ ","
+        print(concat)
```
The function returns the string it doesn't print

isaacrivriv commented on src/element.py:
```diff
@@ -0,0 +1,49 @@
+class Element:
+
+
+    element_dic = {}
```
We dont need to define these here if we set them in the init method so we can remove these

isaacrivriv commented on src/element.py:
```diff
@@ -0,0 +1,49 @@
+
+        #Stores the dictionary with the info for the specific element
+        self.element_dic = PERIODIC_TABLE[a_num]
+        self.atomic_num = self.element_dic['atomic_num']
```
I think its a bit excessive to have them as separate components when we can just have the whole dict saved and access the values as required. We are using double the memory here so we should remove these and just leave the dictionary we get from the table

isaacrivriv reviewed (commented):
Overall looks good. Some comments need addressing before we can merge. After the changes if the class works we should be good to merge

valeriereyes5 committed:
Implemented Review Recommendations

Eliminated the definitions that weren't needed, changed the init method,  fixed string method, eliminated the test

isaacrivriv committed:
Worked on small changes

isaacrivriv commented:
Looks good. Merging
